markdown/events/2022/bytesize-custom_scripts.md

```diff
@@ -22,3 +22,86 @@ It is our hope that these talks / videos will build an archive of training mater
+
+[10:00](https://www.youtube.com/watch?v=3aA5-s8PAF0&t=600)
+This is an example of a mold container.. I mean of how to use dependencies with a custom R script. This is directly from the RNA-seq pipeline. The process is called `salmon_summarizedexperiment`. What you will notice is that down here in the code block, we have `salmon_summarizedexperiment.r`. This is an executable R script with a couple of argument inputs. And at the top here, we have the conda and container declarations. In this situation, it is just a single R packaging tool. This is R base as a part of this package already. You only need to specify this once. And this also already has the Galaxy project and biocontainers, images, containers available. This is a relatively simple example with just the one tool. But you can also have an absolute monster.
+
+[10:59](https://www.youtube.com/watch?v=3aA5-s8PAF0&t=659)
+Also from the RNA-seq pipeline here, we have the `deseq2_qc` process. What you can see here is that we have a very large number of different tools that are implemented as a part of this. In this case, a mold container has been created, which contains all of these. This is probably not a great example because the versions of these tools haven't been pinned to the conda tools. And that's because there were conflicts as this is getting created. But normally, you might expect to see some version numbers after each of these. Again, this is probably a bit more of a monster of a script. I haven't shown this here, but you can find it by going and looking into the RNA-seq repository. And what you can see here is that this is an executable script again with a number of different inputs that can be taken as arguments.
+
+[11:55](https://www.youtube.com/watch?v=3aA5-s8PAF0&t=715)
+Just to summarize, what I've covered today is that NextFlow can use custom scripts written from many different languages. Scripts can be stored in both the bin or the templates directory. And both of these will be available to the Nextflow script. Meaning that you don't need to specify an absolute or a relative path as you're executing a script. It's really fantastic to do this because it makes your scripts much more portable and usable by others. Dependencies can be managed using conda and containers. In the examples I've shown, you can see that it can be quite simple or much more complex with the use of mold containers to help you store all those together with singularity and conda images. And with that, I will finish. I think we're probably about where I thought we'd be for time. And I'll finish on this. And if there are any questions, I'm happy to do my best to answer them. Thanks very much.
```

Comment:
```suggestion
Just to summarize, what I've covered today is that NextFlow can use custom scripts written from many different languages. Scripts can be stored in both the bin or the templates directory. And both of these will be available to the Nextflow script. Meaning that you don't need to specify an absolute or a relative path as you're executing a script. It's really fantastic to do this because it makes your scripts much more portable and usable by others. Dependencies can be managed using conda and containers. In the examples I've shown, you can see that it can be quite simple or much more complex with the use of mulled containers to help you store all those together with singularity and conda images. And with that, I will finish. I think we're probably about where I thought we'd be for time. And I'll finish on this. And if there are any questions, I'm happy to do my best to answer them. Thanks very much.
```

Reply:
mulled containers were used a lot in the talk, maybe we should have another swipe through, to change them all. /FBo

Reply:
Good call! I can't believe I missed so many! Too worried about the audio matching I guess. I've taken another pass at it